api/v1/report.proto

```diff
@@ -18,131 +18,18 @@ package istio.mixer.v1;
 
 option go_package="istio.io/mixer/api/v1;mixerpb";
 
-import "google/protobuf/timestamp.proto";
-import "google/protobuf/struct.proto";
-import "google/protobuf/any.proto";
+import "attributes.proto";
 import "google/rpc/status.proto";
 
 // Used to report telemetry after performing an action.
 message ReportRequest {
   // Uniquely identifies this request, used to match to responses
   int64 request_id = 1;
 
-  // The set of discovered facts associated with this request. This value
-  // represents a delta relative to the facts currently known to the mixer
-  // within the current stream.
-  map<string, string> facts = 2;
-
-  // The monitored resource type associated with this report.
-  string monitored_resource_descriptor_name = 3;
-
-  // Metric values being reported
-  repeated MetricValue metric_values = 4;
-
-  // Log entries being reported
-  repeated LogEntry log_entries = 5;
-}
-
-// Represents a single metric value.
-message MetricValue {
-  // Name of the MetricDescriptor providing the schema for the metric.
-  string descriptor_name = 1;
-
-  // The start of the time period over which this metric value's measurement
-  // applies. The time period has different semantics for different metric
-  // kinds (cumulative, delta, and gauge). See the documentation for the
-  // MetricDescriptor for more information.
-  google.protobuf.Timestamp start_time = 3;
```

Comment:
We aren't supporting aggregation for now, so interval isn't needed. Let's add it back later in a generic way across all attributes?

Although well known attributes that have aggregation intervals would be troublesome...

An alternative to that would be to define:
StringAttribute
BoolAttribute
DoubleAttribute
IntAttribute

Or something akin to that, add the intervals there, and then we can define the well known attributes based on those types, rather than using the raw fields...

Or we go to the suggestion of using enum or string for name, and just have a repeated thing.

Thoughts?